README.md

```diff
@@ -1,46 +1,429 @@
+
+Examples of their use:
+
+#### deployed
+
+Obtain the instance of the contract as deployed by Arc.js:
+
+```javascript
+import { UpgradeScheme } from "@daostack/arc.js";
+const deployedContract = await UpgradeScheme.deployed();
+```
+
+#### new
+
+Call `new()` to migrate a new instance of the contract:
+
+```javascript
+import { UpgradeScheme } from "@daostack/arc.js";
+const newInstance = await UpgradeScheme.new();
+```
+
+#### at
+
+Obtain the wrapper from a given address:
+
+```javascript
+import { UpgradeScheme } from "@daostack/arc.js";
+const newInstance = await UpgradeScheme.at(anAddress);
+```
+
+### Obtain a wrapper from getDeployedContracts
+
+Recall from [Categories of Arc Contracts](#categories-of-arc-contracts) that `getDeployedContracts()` returns categories of Arc contract wrappers.   You can use this information to obtain the wrappers themselves:
+
+```javascript
+import * as ArcJs from '@daostack/arc.js';
+const arcJsWrappers = ArcJs.getDeployedContracts();
+const contractAddress = arcJsWrappers.allContracts.UpgradeScheme.address;
+const contractFactory = arcJsWrappers.allContracts.UpgradeScheme.contract;
+const upgradeScheme = await contractFactory.contract.at(contractAddress);
+```
+
+### Obtain any Arc contract using Utils.requireContract
+
+Not all Arc contracts have been given wrapper classes, for example, `Avatar`, `UController` and many more.  But you can still obtain a raw TruffleContract enabling you to work with these contracts:
+
+```javascript
+import { Utils } from "@daostack/arc.js";
+const truffleContract = await Utils.requireContract("Avatar");
+```
+
+## Working with Arc.js Scripts
+Arc.js contains a set of scripts for building, publishing, running tests and migrating contracts to any network.  These scripts are meant to be accessible and readily usable by client applications.
+
+Typically an application will run an Arc.js script by prefixing `npm explore @daostack/arc.js -- `  to the Arc.js script.  For example, to run the Arc.js script `npm start test.ganache.run` from your application, you would run:
+```
+npm explore @daostack/arc.js -- npm start test.ganache.run
+```
+
+All of the scripts are defined and documented in the package-scripts.js file.  You have already seen their [most typical use in an application context](#setting-up-a-testnet-with-arc-contracts).
+
+See also  [Running Against a Ganache Database](#running-against-a-ganache-database), [Deploying to Other Testnets](#deploying-to-other-testnets) and [Run Lint and Tests](#run-lint-and-tests)
+
+## Deploying to Other Testnets
+The "network" environment variable defines which network arc.js understands you to be deploying to when you run the "migrateContracts" script.  For safety it assumes a different HTTP port for each network. Can be "live" (the mainnet), "kovan" or "ganache".  The default is "ganache".
+
+## Running Against a Ganache Database
```

Comment:
seems like this should come after the first section about running with Ganache

Reply:
I'm not sure which line you are referring to.  In any case, my reasoning is to keep the beginning workflow as simple and straightforward as possible, leaving the special cases until later in the document.

Reply:
+### Setting up a Testnet with Arc Contracts

Reply:
Running against a ganache database isn't the first thing someone needs to learn about, unlike learning to run against ganache in general, which we want to learn right away.  Rather, it falls under the category, for me, of less important special cases that I prefer to leave until later in the document.